File: backend/src/main/kotlin/outcobra/server/service/ExamTaskService.kt
package outcobra.server.service

import outcobra.server.model.dto.ExamTaskDto
import outcobra.server.service.base.BaseService

/**
 * @author Florian Bürgi
 * @since 1.2.0
 */
interface ExamTaskService : BaseService<ExamTaskDto> {
    fun saveAll(examTasks: List<ExamTaskDto>): List<ExamTaskDto>
    fun readByExamId(examId: Long): List<ExamTaskDto>
    fun changeState(examTaskId: Long): ExamTaskDto
}

File: frontend/src/app/exam/service/exam-task.service.ts
import {AppCrudService} from '../../core/services/core/app-crud.service';
import {ExamTaskDto} from '../model/exam.task.dto';
import {Injectable} from '@angular/core';
import {Observable} from 'rxjs/Observable';
import {HttpClient} from '@angular/common/http';

/**
 * @author Florian Bürgi
 * @since 1.2.0
 */
@Injectable()
export class ExamTaskService extends AppCrudService<ExamTaskDto> {
    constructor(http: HttpClient) {
        super(http, "/api/examTask")
    }

    public saveAll(tasks: ExamTaskDto[]): Observable<ExamTaskDto[]> {
        return this._http.put<ExamTaskDto[]>(`/all${this._baseUri}`, tasks);
    }

    public changeState(id: number): Observable<ExamTaskDto> {
        return this._http.post<ExamTaskDto>(`${this._baseUri}/state`, id)
    }
}
